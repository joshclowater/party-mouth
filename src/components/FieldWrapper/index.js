import React from 'react';
import styled from 'styled-components';

const FieldWrapperDiv = styled.div`
  margin-bottom: 2vh;
`;

const Error = styled.span`
  color: red;
`;

const Warning = styled.span`
  color: orange;
`;

export default function FieldWrapper({
  inputName,
  label,
  children,
  touched,
  error,
  warning
}) {
  return (
    <FieldWrapperDiv>
      <label htmlFor={inputName}>{label}</label>
      {children}
      {touched &&
        ((error && <Error>{error}</Error>) ||
          (warning && <Warning>{warning}</Warning>))}
    </FieldWrapperDiv>
  );
}
