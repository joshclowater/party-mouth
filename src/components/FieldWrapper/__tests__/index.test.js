import React from 'react';
import { shallow, mount } from 'enzyme';
import FieldWrapper from '../';

it('Renders correctly', () => {
  expect(
    shallow(
      <FieldWrapper inputName="mock input name" label="mock label">
        "mock children"
      </FieldWrapper>
    )
  ).toMatchSnapshot();
});

describe('touched is true', () => {
  describe('has error and warning', () => {
    const wrapperText = mount(
      <FieldWrapper touched error="mock error" warning="mock warning" />
    ).text();
    it('displays error only', () => {
      expect(wrapperText).toMatch('mock error');
      expect(wrapperText).not.toMatch('mock warning');
    });
  });
  describe('has error', () => {
    const wrapperText = mount(
      <FieldWrapper touched error="mock error" />
    ).text();
    it('displays error', () => {
      expect(wrapperText).toMatch('mock error');
    });
  });
  describe('has only warning', () => {
    const wrapperText = mount(
      <FieldWrapper touched warning="mock warning" />
    ).text();
    it('displays warning', () => {
      expect(wrapperText).toMatch('mock warning');
    });
  });
});

describe('touched is false', () => {
  describe('has error and warning', () => {
    const wrapperText = mount(
      <FieldWrapper touched={false} error="mock error" warning="mock warning" />
    ).text();
    it('displays neither error nor warning', () => {
      expect(wrapperText).not.toMatch('mock error');
      expect(wrapperText).not.toMatch('mock warning');
    });
  });
});
