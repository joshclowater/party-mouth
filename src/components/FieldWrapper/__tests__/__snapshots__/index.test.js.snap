// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<styled.div>
  <label
    htmlFor="mock input name"
  >
    mock label
  </label>
  "mock children"
</styled.div>
`;
